src/components/map/CompassComponent.vue

```diff
@@ -0,0 +1,163 @@
+<script setup lang="ts">
+import type { VcsApp, VcsMap } from '@vcmap/core'
+import { inject, onMounted, ref } from 'vue'
+import IconCompass from '../ui/icons/IconCompass.vue'
+
+const vcsApp = inject('vcsApp') as VcsApp
+
+const compass = ref<HTMLDivElement | null>(null)
+const arrow = ref<HTMLDivElement | null>(null)
+
+onMounted(() => {
+  if (!vcsApp?.maps?.activeMap) {
+    return
+  }
+  syncCompass(vcsApp?.maps?.activeMap)
+})
+
+function syncCompass(map: VcsMap) {
+  return map.postRender.addEventListener(({ map }) => {
+    const vp = map.getViewpointSync()
+    if (vp && vp.isValid()) {
+      transformArrow(vp.pitch)
+      transformNorthPoint(vp.heading)
+    }
+  })
+}
+
+const trackMouse = (
+  // Dirty hack: eslint triggers a unused-vars false positive
+  // eslint-disable-next-line no-unused-vars
+  callback: (e: MouseEvent) => Promise<void>,
+  endCallback: Function = () => {}
+) => {
+  document.body.addEventListener('mousemove', callback)
+  document.body.addEventListener('mouseup', () => {
+    document.body.removeEventListener('mousemove', callback)
+    endCallback()
+  })
+}
+
+function onNorthPointClick() {
+  if (!compass.value || !arrow.value) return
+  const timestamp = Date.now()
+  const clickDuration = 100
+  const { top, left, height, width } = compass.value.getBoundingClientRect()
+
+  const compassPos = {
+    x: left + width / 2,
+    y: top + height / 2,
+  }
+
+  trackMouse(
+    async (e) => {
+      const radians = Math.atan2(
+        compassPos.x - e.clientX,
+        compassPos.y - e.clientY
+      )
+      const angle = (180 / Math.PI) * radians
+      transformNorthPoint(angle)
+      await headingMap(angle)
+    },
+    () => {
+      if (Date.now() - timestamp < clickDuration) {
+        headingMap(0, true)
+      }
+    }
+  )
+}
+
+function onCompassClick() {
+  if (!arrow.value) return
+  const { top, height } = arrow.value.getBoundingClientRect()
+  const yPos = top + height / 2
+
+  trackMouse(async (e) => {
+    const speed = 0.5
+    const initialTilt = -90
```

Comment:
The `initialTilt` here is different compared to the one in `transformArrow`. Could you put a comment to explain it or probably put a comment on what is the reference (e.g. -90 from where or 0 degree is pointing to where)